devops/github-actions/general-docker-deploy.yaml

```diff
@@ -0,0 +1,61 @@
+on:
+  push:
+    branches:
+      - "main"
+      - "dev"
+
+env:
+  FLOWNAME: marvin
+  IMAGE: ${{secrets.IMAGE_REGISTRY_SLASH_IMAGE_REPO}}:latest
+  DEPLOYMENTYAML: ./marvin.yaml
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      # Extract the current branch name to reference Blocks corresponding to this environment
+      - name: Extract branch name
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+        id: extract_branch
+
+      - name: Build the image
+        run: |
+          docker build . --tag ${{ env.IMAGE }} -f ./Dockerfile
+
+      # Replace if needed with plugin in ECR, GCR, GHCR, etc 
+      - name: Login to registry
+        env:
+          CONTAINER_REGISTRY_USER: ${{secrets.CONTAINER_REGISTRY_USER}}
+          CONTAINER_REGISTRY_PASSWORD: ${{secrets.CONTAINER_REGISTRY_PASSWORD}}
+        run: |
+          docker login -u $CONTAINER_REGISTRY_USER -p $CONTAINER_REGISTRY_PASSWORD 
+
+      - name: Push to container registry
+        run: docker push ${{ env.IMAGE }}
+
+      - name: Update Pip, install prefect and required filesystem extras
+        run: pip install -U wheel pip prefect s3fs
+
+      - name: Prefect Cloud login
+        run: |
+          prefect config set PREFECT_API_KEY=$API_KEY
+          prefect config set PREFECT_API_URL=$API_URL
+
+        env:
+          API_KEY: ${{ secrets.PREFECT_API_KEY }}
+          API_URL: ${{ secrets.PREFECT_API_URL }}
```

Comment:
I believe that if you just set the environment variables (`PREFECT_API_KEY` and `PREFECT_API_URL`)  you don't have to run `prefect config set`. So you should be able to get rid of this step and simply pass the variables in the next task (if you want).

Reply:
I was able to get it to work this way, but then you'd have to pass those secrets into `env` for each prefect command 
```
      - name: Build Prefect deployment
        run: prefect deployment build ./prefect/flows/hello.py:$FLOWNAME -n $DEPLOYMENT_NAME -t $TAG -ib $INFRA -sb $STORAGE -o $DEPLOYMENTYAML -q $WORK_QUEUE
        env:
          DEPLOYMENT_NAME: hello-marvin
          TAG: dev
          INFRA: kubernetes-job/default-k8s-job
          STORAGE: s3/flow-script-storage-${{ steps.extract_branch.outputs.branch }}
          WORK_QUEUE: "main"
          PREFECT_API_KEY: ${{ secrets.PREFECT_API_KEY }}
          PREFECT_API_URL: ${{ secrets.PREFECT_API_URL }}
          
      - name: Apply Prefect deployment
        run: prefect deployment apply $DEPLOYMENTYAML
        env:
          PREFECT_API_KEY: ${{ secrets.PREFECT_API_KEY }}
          PREFECT_API_URL: ${{ secrets.PREFECT_API_URL }}
```

imo it seems easier to do it once up front have all subsequent prefect commands just use the config setting like
```
      - name: Prefect Cloud login
        run: |
          prefect config set PREFECT_API_KEY=${{ secrets.PREFECT_API_KEY }}
          prefect config set PREFECT_API_URL=${{ secrets.PREFECT_API_URL }}

      - name: Build Prefect deployment
        run: prefect deployment build ./prefect/flows/hello.py:$FLOWNAME -n $DEPLOYMENT_NAME -t $TAG -ib $INFRA -sb $STORAGE -o $DEPLOYMENTYAML -q $WORK_QUEUE
        env:
          DEPLOYMENT_NAME: hello-marvin
          TAG: dev
          INFRA: kubernetes-job/default-k8s-job
          STORAGE: s3/flow-script-storage-${{ steps.extract_branch.outputs.branch }}
          WORK_QUEUE: "main"
          
      - name: Apply Prefect deployment
        run: prefect deployment apply $DEPLOYMENTYAML
        
```

Reply:
👍🏻

Reply:
ahh that's true, I'll do that